pkg/controller/instance/controller_reconcile.go

```diff
@@ -177,6 +177,12 @@ func (igr *instanceGraphReconciler) handleResourceReconciliation(
 	observed, err := rc.Get(ctx, resource.GetName(), metav1.GetOptions{})
 	if err != nil {
 		if apierrors.IsNotFound(err) {
+			// For read-only resources, we don't create
+			if igr.runtime.ResourceDescriptor(resourceID).IsReadOnly() {
```

Comment:
Can we make sure that the user is only setting name/namespace (+ the resource types).

Reply:
can do.

Reply:
added